src/writer.ts

```diff
@@ -191,16 +201,85 @@ export const handlePropose: CheckpointWriter = async ({ block, tx, rawEvent, eve
   await mysql.queryAsync(query, [item, item.space, user, author]);
 };
 
+export const handleCancel: CheckpointWriter = async ({ rawEvent, event, mysql }) => {
+  if (!rawEvent || !event) return;
+
+  console.log('Handle cancel');
+
+  const space = validateAndParseAddress(rawEvent.from_address);
+  const proposalId = `${space}/${parseInt(event.proposal_id)}`;
+
+  const [{ vote_count }] = await mysql.queryAsync(
+    `SELECT vote_count FROM proposals WHERE id = ? LIMIT 1`,
+    [proposalId]
+  );
+
+  const query = `
+    UPDATE proposals SET cancelled = true WHERE id = ? LIMIT 1;
+    UPDATE spaces SET proposal_count = proposal_count - 1, vote_count = vote_count - ? WHERE id = ? LIMIT 1;
+  `;
+
+  await mysql.queryAsync(query, [proposalId, vote_count, space]);
+};
+
+export const handleUpdate: CheckpointWriter = async ({ block, rawEvent, event, mysql }) => {
+  if (!rawEvent || !event) return;
+
+  console.log('Handle update');
+
+  const space = validateAndParseAddress(rawEvent.from_address);
+  const proposalId = `${space}/${parseInt(event.proposal_id)}`;
+  const metadataUri = longStringToText(event.metadata_URI);
+
+  try {
+    await handleProposalMetadata(metadataUri, mysql);
+
+    const query = `UPDATE proposals SET metadata = ?, edited = ? WHERE id = ? LIMIT 1;`;
```

Comment:
Any reason you do native SQL queries instead of using Checkpoint ORM?

Reply:
I wanted to use ORM, but this PR started as v1 support PR and didn't want to do both in one PR to be able to see starknet v1 support related changes clearer to see. I can convert it to ORM in follow up PR.